### PR TITLE
chore: make shadowgraph compatible with birth rounds

### LINKED ISSUE #11085: Make shadowgraph comaptible with birth rounds
As a prerequisite to https://github.com/hashgraph/hedera-services/issues/10956, make the shadowgraph compatible with the use of birth rounds. This could be done all in one PR, but better to have smaller PRs.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #11085 

This PR makes the shadowgraph compatible with birth rounds. It also includes some shadowgraph related tech debt reduction.

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 1]
```diff
@@ -104,7 +104,7 @@
 import com.swirlds.platform.gossip.GossipFactory;
 import com.swirlds.platform.gossip.IntakeEventCounter;
 import com.swirlds.platform.gossip.NoOpIntakeEventCounter;
-import com.swirlds.platform.gossip.shadowgraph.ShadowGraph;
+import com.swirlds.platform.gossip.shadowgraph.Shadowgraph;
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:107]: Why the change from CamelCase?

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:107]: We always use "shadowgraph" when speaking as if it were a single word, and the camel case always seemed jarring to me. Just an opportunistic cleanup, not strictly required for the core objective of this change.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/Shadowgraph.java [hunk 2]
```diff
@@ -49,187 +50,133 @@
...
             shadow.disconnect();
             shadow.getEvent().clear();
         }
     }
 
     /**
-     * Reserves the events in generation {@code expireBelow}. A reservation prevents events in that generation and later
-     * (higher) generations from being expired from the shadow graph.
+     * Increase the reservation count for the ancient indicator currently held by {@code expireBelow}. A reservation
+     * prevents events that have an ancient indicator not less than the threshold from being expired unlinked.
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/Shadowgraph.java:170]: "from being expired unlinked" grammar

**Review by alittley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/Shadowgraph.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/Shadowgraph.java:170]: fixed

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:107]: We also have `ShadowEvent` and there are other files called `ShadowConfigurationAccessors`.    I see it as two words since there are other `Shadow` things.   Seems like a highly individual perception based on how language operates within them.  You've worked with it long enough that it has become one word to you.  :-)

**Review by edward-swirldslabs — APPROVED** [review]
